File: Java/src/ac/cr/tec/test/MonteCarloTest.java
package ac.cr.tec.test;


import ac.cr.tec.code.MontecarloMethod;
import ac.cr.tec.code.Random;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class MonteCarloTest {

    @Test
    public void distanceTest() {
        double a;
        double b;
        double h;
        a=5;
        b=12;
        h=13;
        Assertions.assertEquals(h, MontecarloMethod.distance(a,b));

        a=7;
        b=24;
        h=25;
        Assertions.assertEquals(h, MontecarloMethod.distance(a,b));

        a=11;
        b=60;
        h=61;
        Assertions.assertEquals(h, MontecarloMethod.distance(a,b));

        a=13;
        b=84;
        h=85;
        Assertions.assertEquals(h, MontecarloMethod.distance(a,b));

        a=17;
        b=144;
        h=145;
        Assertions.assertEquals(h, MontecarloMethod.distance(a,b));

        a=23;
        b=264;
        h=265;
        Assertions.assertEquals(h, MontecarloMethod.distance(a,b));
    }

    @Test
    public void randomTest(){
        int min;
        int max;
        double randomResult;

        min=-2;
        max=13;
        randomResult= Random.random(min,max);
        Assertions.assertTrue(randomResult<=max);
        Assertions.assertTrue(randomResult>=min);


        min=0;
        max=30;
        randomResult= Random.random(min,max);
        Assertions.assertTrue(randomResult<=max);
        Assertions.assertTrue(randomResult>=min);

        min=-13;
        max=-5;
        randomResult= Random.random(min,max);
        Assertions.assertTrue(randomResult<=max);
        Assertions.assertTrue(randomResult>=min);

        min=40;
        max=50;
        randomResult= Random.random(min,max);
        Assertions.assertTrue(randomResult<=max);
        Assertions.assertTrue(randomResult>=min);

        min=-10;
        max=0;
        randomResult= Random.random(min,max);
        Assertions.assertTrue(randomResult<=max);
        Assertions.assertTrue(randomResult>=min);
    }
    @Test
    public void MonteCarloTest(){
        double pi=Math.PI;
        double calculatedPI;
        double diff;
        int iterations=100;
        double allowedMistake=0.00999999;

        while (iterations>0){
            iterations--;
            calculatedPI=MontecarloMethod.computePI();
            System.out.println(calculatedPI);
            diff=Math.abs(calculatedPI-pi);
            Assertions.assertTrue(diff<=allowedMistake);
        }
    }



}

File: Javascript/montecarlo.test.js
 import { calculateDistance,Montecarlo,random } from "./montecarlo.js";

function distanceTest(){
    var a;
    var b;
    var h;
    a=5;
    b=12;
    h=13;
    test('a=5,b=12,c=13',()=>{
        expect(calculateDistance(a,b)).toBe(h);
    });
    
    a=7;
    b=24;
    h=25;
    
    test('a=7,b=24,c=25',()=>{
        expect(calculateDistance(a,b)).toBe(h);
    });
    a=11;
    b=60;
    h=61;
    test('a=11,b=60,c=61',()=>{
        expect(calculateDistance(a,b)).toBe(h);
    });
   
    a=13;
    b=84;
    h=85;
    test('a=13,b=84,c=85',()=>{
        expect(calculateDistance(a,b)).toBe(h);
    });

    a=17;
    b=144;
    h=145;
    test('a=17,b=144,c=145',()=>{
        expect(calculateDistance(a,b)).toBe(h);
    });
   
    a=23;
    b=264;
    h=265;
    test('a=23,b=264,c=265',()=>{
        expect(calculateDistance(a,b)).toBe(h);
    });
   
}

function randomTest(){
    var min;
    var max;
    var randomValue;
 
 
    min=-10;
    max=10;
    randomValue=random(min,max);
    test("min=-10, max=10",()=>{
        expect(randomValue).toBeGreaterThanOrEqual(min);
        expect(randomValue).toBeLessThanOrEqual(max);
 
    });
 
 
    min=-2;
    max=13;
    randomValue=random(min,max);
    test("min=-2, max=13",()=>{
        expect(randomValue).toBeGreaterThanOrEqual(min);
        expect(randomValue).toBeLessThanOrEqual(max);
 
    });
 
 
 
    min=0;
    max=30;
    randomValue=random(min,max);
    test("min=0, max=30",()=>{
        expect(randomValue).toBeGreaterThanOrEqual(min);
        expect(randomValue).toBeLessThanOrEqual(max);
 
    });
 
 
    min=-13;
    max=-5;
    randomValue=random(min,max);
    test("min=-13, max=-5",()=>{
        expect(randomValue).toBeGreaterThanOrEqual(min);
        expect(randomValue).toBeLessThanOrEqual(max);
 
    });
 
 
    min=40;
    max=50;
    randomValue=random(min,max);
    test("min=40, max=50",()=>{
        expect(randomValue).toBeGreaterThanOrEqual(min);
        expect(randomValue).toBeLessThanOrEqual(max);
 
    });
 
 
    min=-10;
    max=0;
    randomValue=random(min,max);
    test("min=-10, max=0",()=>{
        expect(randomValue).toBeGreaterThanOrEqual(min);
        expect(randomValue).toBeLessThanOrEqual(max);
 
    });
   
}
